sdk/python/kfp/registry/client.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class for KFP Registry Client."""
+
+import logging
+
+import google.auth
+import json
+import requests
+import re
+from google.auth.credentials import Credentials
+from typing import Any, Optional, List, Tuple, Union
+
+_KNOWN_HOSTS_REGEX = {
+    "kfp_pkg_dev":
+        r'(^https\:\/\/(?P<location>[\w\-]+)\-kfp\.pkg\.dev\/(?P<project_id>.*)\/(?P<repo_id>.*))',
+}
+
+_DEFAULT_JSON_HEADER = {
+    'Content-type': 'application/json',
+}
+
+
+class _SafeDict(dict):
+
+    def __missing__(self, key: str):
+        return '{' + key + '}'
+
+
+class ApiAuth(requests.auth.AuthBase):
+
+    def __init__(self, token: str):
+        self._token = token
+
+    def __call__(self, request: requests.Request):
+        request.headers['authorization'] = 'Bearer ' + self._token
+        return request
+
+
+class RegistryClient:
```

Comment:
Can you add docstrings for the `RegistryClient`? I don’t think we necessarily need to provide docstrings for each method (they are fairly self-explanatory), but we probably should for the class itself.

Reply:
We should add docstrings for each public methods, even if they are self-explanatory. 
They would be useful for generating the doc. For example: https://kubeflow-pipelines.readthedocs.io/en/stable/source/kfp.client.html

Reply:
Done.